### PR TITLE
20200820 inplace edit

### PR DESCRIPTION
**Pull request by changlong-liu**

### CODE REVIEW

File: src/plugins/azgenerator/AzExtensionIncrementalGenerator.ts [hunk 1]
```diff
@@ -40,6 +41,9 @@ export class AzExtensionIncrementalGenerator extends AzGeneratorBase {
         this.files[path.join(this.azDirectory, PathConstants.generatedFolder, PathConstants.initFile)] = GenerateNamespaceInit(this.model);
 
         this.files[path.join(this.azDirectory, PathConstants.testFolder, PathConstants.latestFolder, PathConstants.incTestScenarioFile(this.model.Extension_NameUnderscored))] = GenerateAzureCliTestScenario(this.model);
+        let testFile = PathConstants.testFolder + "/" + PathConstants.latestFolder + "/" + PathConstants.incTestScenarioFile(this.model.Extension_NameUnderscored);
+        this.files[path.join(this.azDirectory, testFile)] = inplaceGen(path.join(this.model.CliOutputFolder, this.azDirectory), testFile, GenerateAzureCliTestScenario(this.model));
```
Comment (LianwMS) [inline src/plugins/azgenerator/AzExtensionIncrementalGenerator.ts:45]: Consider to refactor GenerateAzureCliTestScenario to implement TemplateBase
And use generateIncrementalSingleAndAddtoOutput or generateFullSingleAndAddtoOutput to replace the dup logic here. current logic is dup in CoreFull CoreIncre ExtFull and ExtIncre generators.

File: src/plugins/azgenerator/CodeModelAzImpl.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ class ActionParam {
 export class CodeModelCliImpl implements CodeModelAz {
     codeModel: CodeModel;
     options: any;
+    outputFolder: string;
```
Comment (LianwMS) [inline src/plugins/azgenerator/CodeModelAzImpl.ts:24]: where is the code to use outputFolder paraeter?

File: src/test/scenarios/synapse/output/coredefault/src/azure-cli/requirements.py3.Darwin.txt [hunk 3]
```diff
@@ -127,4 +127,3 @@ wcwidth==0.1.7
 websocket-client==0.56.0
 wrapt==1.11.2
 xmltodict==0.12.0
-azure-mgmt-synapse==0.3.0
```
Comment (LianwMS) [inline src/test/scenarios/synapse/output/coredefault/src/azure-cli/requirements.py3.Darwin.txt:130]: why this package will not be installed after change?

**Review by LianwMS — COMMENTED** [review]
Rebase the master first

**Comment by changlong-liu** [comment]
Hi @LianwMS , refactored in latest commits

File: src/plugins/azgenerator/CodeModelAzImpl.ts (hunk 2 above)
Comment (changlong-liu) [reply src/plugins/azgenerator/CodeModelAzImpl.ts]: removed in latest commit

**Review by LianwMS — APPROVED** [review]
:shipit:

**Pull request merged**